sys/isrpipe/isrpipe.c

```diff
@@ -0,0 +1,29 @@
+#include "isrpipe.h"
+
+void isrpipe_init(isrpipe_t *isrpipe, char *buf, size_t bufsize)
+{
+    mutex_init(&isrpipe->mutex);
+    tsrb_init(&isrpipe->tsrb, buf, bufsize);
+}
+
+int isrpipe_write_one(isrpipe_t *isrpipe, char c)
+{
+    int res = tsrb_add_one(&isrpipe->tsrb, c);
+
+    /* `res` is either 0 on success or -1 when the buffer is full. Either way,
+     * unlocking the mutex is fine.
+     */
+    mutex_unlock(&isrpipe->mutex);
+
+    return res;
+}
+
+int isrpipe_read(isrpipe_t *isrpipe, char *buffer, size_t count)
+{
+    int res;
+
+    while (!(res = tsrb_get(&isrpipe->tsrb, buffer, count))) {
```

Comment:
@lebrush: I think for the context of stdio_uart that 'single byte read' behavior is intended, as we never know how many byte we actually expect... But for other use-cases I think this could be improved?!

@kaspar030: have you spend thought also on on functions for `return once N byte are available` as described above, and on `return once last received byte equals SOMEVAL`? The latter could be quite useful for the stdio implementation waiting for `\n` chars...